extra_data/write_cxi.py

```diff
@@ -82,10 +79,10 @@ def collect_pulse_ids(self):
         pulse_ids = np.full((self.nframes, self.nmodules), NO_PULSE_ID,
                             dtype=np.uint64)
 
+        pulse_key = self.group_label + '.' + self.pulse_id_label
         for source, modno in self.detdata.source_to_modno.items():
-            module_id = self.modulenos.index(modno)
-            for chunk in self.data._find_data_chunks(source,
-                                                     self.pulse_id_label):
+            module_id = self._get_module_id(modno)
```

Comment:
It would be good to have the method in the base class, even if it raises NotImplementedError, because otherwise my IDE complains that it's calling something that doesn't exist:

![Screenshot from 2021-05-31 11-42-26](https://user-images.githubusercontent.com/327925/120181747-66c8e900-c205-11eb-8fc5-a4e0f7180eae.png)

Also, I don't think 'module ID' is a great name, when it's going from the numbers people use to identify the modules to something else. Maybe call it 'module index' instead? index can be abbreviated 'ix' if you need to.

Reply:
Implemented in https://github.com/European-XFEL/EXtra-data/commit/6582e9954511eb0bc752851d8f64a31bb7ed1348 and https://github.com/European-XFEL/EXtra-data/commit/2758b23ae00008c8ca120cdc6ee6ae01cb6a04dd.